### PR TITLE
[textanalytics] Add Debug Proxy class for ExtractKeyPhrasesResultCollection

### LINKED ISSUE #16283: Improve how we surface in the debugger properties that are in xxCollection types so they don't get lost in the Raw view area.
Examples are:
- [PiiEntityCollection.RedactedText](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/textanalytics/Azure.AI.TextAnalytics/api/Azure.AI.TextAnalytics.netstandard2.0.cs#L353)
- All [model versions](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/textanalytics/Azure.AI.TextAnalytics/api/Azure.AI.TextAnalytics.netstandard2.0.cs#L46)
- All statistics

### PR DESCRIPTION
**Pull request by a-noyass**
fixes https://github.com/Azure/azure-sdk-for-net/issues/16283

# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [ ] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/master/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running generate.ps1. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version,

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

**Comment by a-noyass** [comment]
@maririos So this is the only way I could get it to work. Adding a new class that defines how the Collection will look like in the debugger. If this works I can add do it for the other classes as well, or maybe this can be done during generation somehow.
The result looks like this 
Screenshot 2021-02-11 100900

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/ExtractKeyPhrasesResultCollectionDebugView.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Azure.AI.TextAnalytics
+{
+    /// <summary>
+    /// Debugger Proxy class for <see cref="ExtractKeyPhrasesResultCollection"/>.
+    /// </summary>
+    public class ExtractKeyPhrasesResultCollectionDebugView
```
Comment (maririos) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/ExtractKeyPhrasesResultCollectionDebugView.cs:12]: we can make this class internal

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/ExtractKeyPhrasesResultCollectionDebugView.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    {
+        private ExtractKeyPhrasesResultCollection BaseCollection { get; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ExtractKeyPhrasesResultCollectionDebugView"/> class.
+        /// </summary>
+        /// <param name="collection"></param>
+        public ExtractKeyPhrasesResultCollectionDebugView(ExtractKeyPhrasesResultCollection collection)
```
Comment (maririos) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/ExtractKeyPhrasesResultCollectionDebugView.cs:20]: we can follow what https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/src/Internal/ResponseDebugView.cs is doing. We won't need to add the docstrings so no worries about later keeping them up to date

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/ExtractKeyPhrasesResultCollectionDebugView.cs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        {
+            BaseCollection = collection;
+        }
+
+        /// <summary>
+        /// Collection of <see cref="ExtractKeyPhrasesResult"/> objects corresponding
+        /// to a batch of documents.
+        /// </summary>
+        public List<ExtractKeyPhrasesResult> Items
```
Comment (maririos) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/ExtractKeyPhrasesResultCollectionDebugView.cs:29]: my concern with having `Items` is that the user might see this in the debugger and then they will expect for `ExtractKeyPhrasesResultCollection` to have a property called `Items`.

**Comment by maririos** [comment]
@pakrym do you know if there is another way we could achieve showing `Model Version` and `Statistics` in the same level as the items of the collection?
something like:
https://user-images.githubusercontent.com/9868623/107310059-7ba82380-6a40-11eb-8d5a-6a830d95c86e.png

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/ExtractKeyPhrasesResultCollectionDebugView.cs (hunk 3 above)
Comment (pakrym) [reply sdk/textanalytics/Azure.AI.TextAnalytics/src/ExtractKeyPhrasesResultCollectionDebugView.cs:29]: Add `[DebuggerBrowsable(DebuggerBrowsableState.RootHidden)]` on this property.

**Comment by pakrym** [comment]
Yeah, try using `[DebuggerBrowsable(DebuggerBrowsableState.RootHidden)]` on the `Items` property.

**Comment by a-noyass** [comment]
@maririos I moved the DebugView class inside the original class like we discussed before so it's not forgotten when changes are made to the original class.
I also tried the DebuggerBrowsable attribute and it worked but this way we have two Raw View fields in the debugger one for items and one for the main class which seems a bit confusing. What do you think?
image

**Comment by maririos** [comment]
I like it. I think seeing the Model Version and Statistics there is more important. @suhas92 and @abhahn this address the feedback we got in latest UX Study. What do you think?

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/PiiEntityCollection.cs [hunk 4]
```diff
@@ -29,5 +32,35 @@ internal PiiEntityCollection(IList<PiiEntity> entities, string redactedText, ILi
         /// Warnings encountered while processing the document.
         /// </summary>
         public IReadOnlyCollection<TextAnalyticsWarning> Warnings { get; }
+
+        /// <summary>
+        /// Debugger Proxy class for <see cref="PiiEntityCollection"/>.
+        /// </summary>
+        internal class PiiEntityCollectionDebugView
```
Comment (maririos) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/PiiEntityCollection.cs:39]: missing `RedactedText` property

**Review by maririos — APPROVED** [review]

**Comment by suhas92** [comment]
> I like it. I think seeing the Model Version and Statistics there is more important. @suhas92 and @abhahn this address the feedback we got in latest UX Study. What do you think?

@maririos Looks good to me.

**Review by suhas92 — APPROVED** [review]

**Pull request merged**